File: app/src/main/java/com/se/music/fragment/LocalAlbumFragment.java
package com.se.music.fragment;

import android.os.AsyncTask;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.facebook.drawee.view.SimpleDraweeView;
import com.se.music.R;
import com.se.music.base.BaseFragment;
import com.se.music.common.entity.AlbumEntity;
import com.se.music.utils.MusicUtils;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class LocalAlbumFragment extends BaseFragment {

    @BindView(R.id.local_album_recycle)
    RecyclerView mRecyclerView;

    private List<AlbumEntity> list = new ArrayList<>();

    private LocalAlbumFragmentAdapter mAdapter;

    public LocalAlbumFragment() {
    }

    public static LocalAlbumFragment newInstance() {
        LocalAlbumFragment fragment = new LocalAlbumFragment();
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_local_album, container, false);
        ButterKnife.bind(this, view);
        mAdapter = new LocalAlbumFragmentAdapter();
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setAdapter(mAdapter);
        return view;
    }

    //更新adapter界面
    @Override
    public void reloadAdapter() {
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(final Void... unused) {
                list.addAll(MusicUtils.Companion.queryAlbums(getContext()));
                return null;
            }

            @Override
            protected void onPostExecute(Void aVoid) {
                mAdapter.notifyDataSetChanged();
            }
        }.execute();
    }

    private class LocalAlbumFragmentAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

        @Override
        public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            return new SingerItemViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.singer_item, parent, false));
        }

        @Override
        public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
            ((SingerItemViewHolder) holder).onBindData(list.get(position));
        }

        @Override
        public int getItemCount() {
            return list.size();
        }
    }

    class SingerItemViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.img_singer)
        SimpleDraweeView simpleDraweeView;

        @BindView(R.id.singer_list_title)
        TextView mTitle;

        @BindView(R.id.singer_list_info)
        TextView mInfo;

        public SingerItemViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

        @OnClick(R.id.rl_singer_item)
        void click() {
        }

        public void onBindData(AlbumEntity albumEntity) {
            mTitle.setText(albumEntity.getAlbumName());
            mInfo.setText(albumEntity.getNumberOfSongs() + "首");
            simpleDraweeView.setImageURI(albumEntity.getAlbumArt() + "");
        }
    }
}

File: app/src/main/java/com/se/music/fragment/LocalAlbumFragment.kt
package com.se.music.fragment

import android.database.Cursor
import android.os.Bundle
import android.support.v4.app.LoaderManager
import android.support.v4.content.CursorLoader
import android.support.v4.content.Loader
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.se.music.R
import com.se.music.adapter.AlbumListAdapter
import com.se.music.base.BaseFragment
import com.se.music.entity.AlbumEntity
import com.se.music.provider.metadata.albumSelection
import com.se.music.provider.metadata.info_album
import com.se.music.provider.metadata.localAlbumUri
import com.se.music.utils.QUERY_LOCAL_ALBUM
import com.se.music.utils.parseCursorToAlbumEntityList

/**
 *Author: gaojin
 *Time: 2018/7/8 下午5:21
 */

class LocalAlbumFragment : BaseFragment(), LoaderManager.LoaderCallbacks<Cursor> {
    private lateinit var mRecyclerView: RecyclerView
    private lateinit var adapter: AlbumListAdapter
    private val list = ArrayList<AlbumEntity>()

    companion object {
        fun newInstance(): LocalAlbumFragment {
            return LocalAlbumFragment()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        mRecyclerView = inflater.inflate(R.layout.fragment_local_recycler_view, container, false) as RecyclerView
        return mRecyclerView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        adapter = AlbumListAdapter(context!!, list, loaderManager)
        mRecyclerView.layoutManager = LinearLayoutManager(activity)
        mRecyclerView.setHasFixedSize(true)
        mRecyclerView.adapter = adapter
        loaderManager.initLoader(QUERY_LOCAL_ALBUM, null, this)
    }

    override fun onCreateLoader(id: Int, args: Bundle?): Loader<Cursor> {
        return CursorLoader(context!!
                , localAlbumUri
                , info_album
                , albumSelection.toString()
                , null
                , null)
    }

    override fun onLoadFinished(loader: Loader<Cursor>, data: Cursor) {
        parseCursorToAlbumEntityList(QUERY_LOCAL_ALBUM, data, list)
        adapter.notifyDataSetChanged()
    }

    override fun onLoaderReset(loader: Loader<Cursor>) {
    }
}